tests/units/cli/test_default_provider.py

```diff
@@ -0,0 +1,76 @@
+"""
+ Copyright 2015 Red Hat, Inc.
+
+ This file is part of Atomic App.
+
+ Atomic App is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Lesser General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ Atomic App is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public License
+ along with Atomic App. If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import os
+import sys
+import pytest
+
+import atomicapp.cli.main
+
+
+class TestCli(object):
+    # We're using object instead of UnitTest for this in order to use capsys effectively
+
+    def exec_cli(self, command):
+        saved_args = sys.argv
+        sys.argv = command
+        atomicapp.cli.main.main()
+        sys.argv = saved_args
+
+    def set_up(self):
+        self.examples_dir = os.path.dirname(__file__) + '/test_examples/'
+
+    def tear_down(self):
+        top = os.path.dirname(__file__) + '/test_examples/'
+        for root, dirs, files in os.walk(top):
+            for f in files:
+                if f.startswith('.'):
+                    os.remove(os.path.join(root, f))
+                elif f == "answers.conf.gen":
+                    os.remove(os.path.join(root, f))
+
+    def test_run_helloapache_app(self, capsys):
+        # Let's set this example up!
+        self.set_up()
+
+        # Prepare the CLI arguments
+        command = [
+            "main.py",
+            "--verbose",
+            "--dry-run",
+            "run",
+            self.examples_dir + 'oneprovider-helloapache/'
+        ]
+
+        # Run the dry-run command
+        with pytest.raises(SystemExit) as exec_info:
```

Comment:
I know that ``pytest`` has been included as a test dependency quite some time back, but are we using ``pytest`` apart from ``pytest.raises``. Python's unittest.TestCase provides ``assertRaises`` method to do similar stuff. Any thoughts?